PATCH-P1-FIX-UPSTREAM: Fix mysqld linking

From 2e9a7f69ac3c685684696a7a2dc26b17f3e5a606 Mon Sep 17 00:00:00 2001
From: Kristyna Streitova <kstreitova@suse.com>
Date: Thu, 25 Jun 2015 10:18:10 +0200
Subject: [PATCH] Use CMAKE STATIC declaration where needed

This prevents broken build with -DSHARED enforced on cmake command line
---
 libservices/CMakeLists.txt | 2 +-
 sql/CMakeLists.txt         | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/libservices/CMakeLists.txt b/libservices/CMakeLists.txt
index c0bdadb..246ae7d 100644
--- a/libservices/CMakeLists.txt
+++ b/libservices/CMakeLists.txt
@@ -28,5 +28,5 @@ SET(MYSQLSERVICES_SOURCES
   rpl_transaction_ctx_service.c
   rpl_transaction_write_set_service.c)
 
-ADD_LIBRARY(mysqlservices ${MYSQLSERVICES_SOURCES})
+ADD_LIBRARY(mysqlservices STATIC ${MYSQLSERVICES_SOURCES})
 INSTALL(TARGETS mysqlservices DESTINATION ${INSTALL_LIBDIR} COMPONENT Development)
diff --git a/sql/CMakeLists.txt b/sql/CMakeLists.txt
index 58a32cc..7feb6de 100644
--- a/sql/CMakeLists.txt
+++ b/sql/CMakeLists.txt
@@ -355,15 +355,15 @@ SET (BINLOG_SOURCE rpl_gtid_misc.cc
                    log_event.cc log_event_old.cc binlog.cc sql_binlog.cc
                    rpl_filter.cc rpl_record.cc rpl_record_old.cc
                    rpl_utility.cc rpl_injector.cc rpl_table_access.cc)
-ADD_LIBRARY(binlog ${BINLOG_SOURCE})
+ADD_LIBRARY(binlog STATIC ${BINLOG_SOURCE})
 TARGET_LINK_LIBRARIES(binlog binlogevents_static)
 
 SET (RPL_SOURCE rpl_handler.cc rpl_tblmap.cc rpl_context.cc)
 ADD_DEPENDENCIES(binlog GenError)
-ADD_LIBRARY(rpl ${RPL_SOURCE})
+ADD_LIBRARY(rpl STATIC ${RPL_SOURCE})
 SET (MASTER_SOURCE rpl_master.cc rpl_binlog_sender.cc)
 ADD_DEPENDENCIES(rpl GenError)
-ADD_LIBRARY(master ${MASTER_SOURCE})
+ADD_LIBRARY(master STATIC ${MASTER_SOURCE})
 ADD_DEPENDENCIES(master GenError)
 SET (SLAVE_SOURCE rpl_slave.cc rpl_reporting.cc rpl_mi.cc rpl_rli.cc
 		  rpl_info_handler.cc rpl_info_file.cc
@@ -372,7 +372,7 @@ SET (SLAVE_SOURCE rpl_slave.cc rpl_reporting.cc rpl_mi.cc rpl_rli.cc
                   rpl_rli_pdb.cc rpl_info_dummy.cc rpl_mts_submode.cc
                   rpl_slave_commit_order_manager.cc rpl_msr.cc
                   rpl_trx_boundary_parser.cc rpl_channel_service_interface.cc)
-ADD_LIBRARY(slave ${SLAVE_SOURCE})
+ADD_LIBRARY(slave STATIC ${SLAVE_SOURCE})
 ADD_DEPENDENCIES(slave GenError)
 
 ######################### GUnit Lib #################################
